__tests__/components/JellyForm.spec.js

```diff
@@ -0,0 +1,398 @@
+    `
+
+    mount(
+      <Provider>
+        <JellyForm schema={uuidSchema} />
+      </Provider>
+    )
+  })
+
+  describe('hideSubmitButton property', () => {
+    it('should render a submit button if the property is not specified', () => {
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(1)
+
+      component.unmount()
+    })
+
+    it('should render a submit button if the property is falsey', () => {
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} hideSubmitButton={false} />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(1)
+
+      component.unmount()
+    })
+
+    it('should not render a submit button if the property is truthey', () => {
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} hideSubmitButton />
+        </Provider>
+      )
+
+      expect(component.find('button')).toHaveLength(0)
+
+      component.unmount()
+    })
+  })
+
+  describe('submitButtonText property', () => {
+    it('should change the text value of the submit button', () => {
+      const submitText = 'Click me to submit this form'
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} submitButtonText={submitText} />
+        </Provider>
+      )
+
+      expect(component.find('button').text()).toEqual(submitText)
+
+      component.unmount()
+    })
+
+    it('should accept a JSX element as a value', () => {
+      const submitText = 'Click me to submit this form'
+      const submitElement = <span>{submitText}</span>
+
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} submitButtonText={submitElement} />
+        </Provider>
+      )
+
+      expect(component.find('button').text()).toEqual(submitText)
+
+      component.unmount()
+    })
+  })
+
+  describe('submitButtonProps property', () => {
+    it('should set submit button props', () => {
+      const submitButtonProps = {
+        className: 'custom-button-class'
+      }
+
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} submitButtonProps={submitButtonProps} />
+        </Provider>
+      )
+
+      expect(component.find('button').hasClass(submitButtonProps.className)).toEqual(true)
+
+      component.unmount()
+    })
+  })
+
+  describe('onFormSubmit property', () => {
+    it('should be called when the submit button is clicked', () => {
+      const value = 'Squirtle'
+      const callback = sinon.spy()
+      const component = mount(
+        <Provider>
+          <JellyForm schema={schema} onFormSubmit={callback} />
+        </Provider>
+      )
+
+      const input = component.find('input')
+      input.simulate('change', { target: { value } })
+      component.update()
+
+      return Promise.delay(150).then(() => {
```

Comment:
The documentation you link to is for ShallowWrapper which is not whats being used here. I understand your concern about flaky tests, but in this particular case, the tests are not flaky and have ran many many times without issue. I'll evaluate which of these delays are required, it may be that some of them can be removed.